src/project_root.rs

```diff
@@ -17,10 +17,11 @@ pub fn roots_by_marker(roots: &[String], path: &str) -> String {
     while !pwd.is_dir() {
         pwd.pop();
     }
-    let src = pwd.to_str().unwrap().to_string();
+    let src = pwd.clone();
 
-    loop {
-        for root in roots {
+    for root in roots {
+        pwd = src.clone();
```

Comment:
nit: I think we can get rid of extra variable and one extra clone:

```diff
@@ -17,10 +17,9 @@ pub fn roots_by_marker(roots: &[String], path: &str) -> String {
     while !pwd.is_dir() {
         pwd.pop();
     }
-    let src = pwd.clone();
 
     for root in roots {
-        pwd = src.clone();
+        let mut pwd = pwd.clone();
         loop {
             // unwrap should be safe here because we walk up path previously converted from str
             let matches = glob(pwd.join(root).to_str().unwrap());
@@ -31,11 +30,11 @@ pub fn roots_by_marker(roots: &[String], path: &str) -> String {
                 }
             }
             if !pwd.pop() {
-              break;
+                break;
             }
         }
     }
-    return src.to_str().unwrap().to_string();
+    return pwd.to_str().unwrap().to_string();
 }
```

If find shadowing confusing here, we can choose a different name for the `pwd` in the loop.